### ISSUE TITLE
Add Mermaid support to Docusaurus v2

### ISSUE THREAD

**Issue by sideeffffect** [zio/zio-prelude#780]
_Difficulty: Beginner, prior experience with Docusaurus or yarn/npm is an advantage_

We have some diagram in Mermaid https://github.com/zio/zio-prelude/blob/master/docs/overview/diagrams.md
It would be nice to have them rendered to pictures automatically.

Docusaurus v2 can do that https://github.com/facebook/docusaurus/issues/1258#issuecomment-594393744

blocked on https://github.com/zio/zio-prelude/issues/779
